app/controllers/api_controller/vms.rb

```diff
@@ -250,5 +260,13 @@ def vm_event(vm, event_type, event_message, event_time)
     rescue => err
       action_result(false, err.to_s)
     end
+
+    def retire_vm(vm, id, data)
+      desc = "#{vm_ident(vm)} retiring"
+      result = retire_resource(:vms, id, data)
+      action_result(true, desc, :result => result)
```

Comment:
with action results, let's not return the updated vm, unless we really have to.

maybe update with:

```
desc = "#{vm_ident(vm)} retiring"
desc << " on #{data['date']}" if Hash(data)['date'].present?
result = retire_resource(:vms, id, data)
action_result(true, desc)
```

updating the specs accordingly.